apps/aecore/src/aec_tx_pool_sync.erl

```diff
@@ -0,0 +1,477 @@
+    get_subtree(Path, Tree);
+unfold({key, Key}, _Tree) ->
+    [{key, Key}].
+
+get_subtree(Key, _Tree) when bit_size(Key) =:= 32 * 8 ->
+    [{key, Key}];
+get_subtree(Path, Tree) ->
+    get_subtree(aeu_mp_trees:iterator_from(Path, Tree), Path, bit_size(Path), []).
+
+get_subtree(Iter, Path, S, Acc) ->
+    case aeu_mp_trees:iterator_next(Iter) of
+        {Key = <<Path:S/bits, _Rest/bits>>, _, NewIter} ->
+            get_subtree(NewIter, Path, S, [{key, Key} | Acc]);
+        _X ->
+            lists:reverse(Acc)
+    end.
+
+build_tx_mpt() ->
+    try
+        F = fun(TxHash, Tree) ->
+                aeu_mp_trees:put(TxHash, [], Tree)
+            end,
+        Tree = aec_db:fold_mempool(F, aeu_mp_trees:new()),
+        {ok, Tree}
+    catch _:R ->
+        {error, R}
+    end.
+
+-define(NODE_TAG, 0).
+-define(LEAF_TAG, 1).
+-define(SUBTREE_TAG, 2).
+-define(KEY_TAG, 3).
+
+serialize_unfolds(Us) ->
+    [ serialize_unfold(U) || U <- Us ].
+
+deserialize_unfolds(SUs) ->
+    [ deserialize_unfold(SU) || SU <- SUs ].
+
+serialize_unfold({node, Path, Node}) ->
+    aeu_rlp:encode(
+        aec_serialization:encode_fields(
+            [{type, int}, {path, binary}, {node, binary}],
+            [{type, ?NODE_TAG}, {path, serialize_niblets(Path)}, {node, Node}]));
+serialize_unfold({leaf, Path}) ->
+    aeu_rlp:encode(
+        aec_serialization:encode_fields(
+            [{type, int}, {leaf, binary}],
+            [{type, ?LEAF_TAG}, {leaf, serialize_niblets(Path)}]));
+serialize_unfold({subtree, Path}) ->
+    aeu_rlp:encode(
+        aec_serialization:encode_fields(
+            [{type, int}, {subtree, binary}],
+            [{type, ?SUBTREE_TAG}, {subtree, serialize_niblets(Path)}]));
+serialize_unfold({key, Key}) ->
+    aeu_rlp:encode(
+        aec_serialization:encode_fields(
+            [{type, int}, {key, binary}],
+            [{type, ?KEY_TAG}, {key, Key}])).
+
+deserialize_unfold(Blob) ->
+    try
+        [TypeBin | Fields] = aeu_rlp:decode(Blob),
+        [{type, Type}] = aec_serialization:decode_fields([{type, int}], [TypeBin]),
+        deserialize_unfold(Type, Fields)
+    catch _:Reason ->
+        {error, Reason}
+    end.
+
+deserialize_unfold(?NODE_TAG, Flds) ->
+    [{path, Path}, {node, Node}] =
+        aec_serialization:decode_fields([{path, binary}, {node, binary}], Flds),
+    {node, deserialize_niblets(Path), Node};
+deserialize_unfold(?LEAF_TAG, Flds) ->
+    [{leaf, Path}] = aec_serialization:decode_fields([{leaf, binary}], Flds),
+    {leaf, deserialize_niblets(Path)};
+deserialize_unfold(?SUBTREE_TAG, Flds) ->
+    [{subtree, Path}] = aec_serialization:decode_fields([{subtree, binary}], Flds),
+    {subtree, deserialize_niblets(Path)};
+deserialize_unfold(?KEY_TAG, Flds) ->
+    [{key, Key}] = aec_serialization:decode_fields([{key, binary}], Flds),
+    {key, Key}.
+
+%% Paths are not necessarily even bytes, this is required by RLP so we have to
+%% pad this using parity information.
```

Comment:
Internally in the `aeu_mp_trees.erl` this is handled here:

https://github.com/aeternity/epoch/blob/aa2017a6e94164ae19940565c29aea4b39a32b12/apps/aeutils/src/aeu_mp_trees.erl#L831